internal/distro/fedora/images.go

```diff
@@ -143,6 +143,31 @@ func osCustomizations(
 		// should have been validated before this point.
 		panic(fmt.Sprintf("failed to convert file customizations to fs node files: %v", err))
 	}
+
+	// since we are replacing the inline gpgkeys with their file paths,
+	// we need to make a copy of the repos so we don't modify the original
+	customRepos := blueprint.DeepCopyCustomRepositories(c.GetRepositories())
+
+	// get the inline gpg keys from the repos and convert them to fs node files
+	// so we can save them to disk. This step also swaps the inline gpg key
+	// with the path to the file in the os file tree
+	gpgKeyFiles, err := blueprint.GetInlineGPGKeysFromCustomRepositories(customRepos)
+	if err != nil {
+		panic(fmt.Sprintf("failed to convert inline gpgkeys to fs node files: %v", err))
+	}
```

Comment:
As I commented elsewhere, I propose to get `gpgKeyFiles` and `yumRepos` in a single function call without the need to make a copy of the customizations.